### PR TITLE
Bugfix/2439 Fix OpenFile bottom sheet dismiss issue on swiping

### LINKED ISSUE #2439: Scrolling up in "open with" app selection bottom sheet is very difficult
Scrolling up in "open with" app selection bottom sheet is very difficult

### PR DESCRIPTION
**Pull request by VishalNehra**
#### Issue tracker   

Fixes #2439 

#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [x] Done  
  

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFileDialogFragment.kt [hunk 1]
```diff
@@ -317,6 +319,30 @@ class OpenFileDialogFragment : BaseBottomSheetFragment() {
...
     }
 
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState) as BottomSheetDialog
+
+        dialog.setOnShowListener {
+            val bottomSheet = (it as BottomSheetDialog)
+                .findViewById<View>(com.google.android.material.R.id.design_bottom_sheet)
+                as FrameLayout?
+            val behavior = BottomSheetBehavior.from(bottomSheet!!)
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFileDialogFragment.kt:329]: Please remove !!

File: app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFileDialogFragment.kt [hunk 2]
```diff
@@ -317,6 +319,30 @@ class OpenFileDialogFragment : BaseBottomSheetFragment() {
         viewBinding.appsListView.setOnScrollListener(preloader)
     }
 
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState) as BottomSheetDialog
+
+        dialog.setOnShowListener {
+            val bottomSheet = (it as BottomSheetDialog)
+                .findViewById<View>(com.google.android.material.R.id.design_bottom_sheet)
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFileDialogFragment.kt:327]: Please fix findViewById<View> with a cast later.

**Review by EmmanuelMess — CHANGES_REQUESTED** [review]
See comments

File: app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFileDialogFragment.kt [hunk 3]
```diff
@@ -324,21 +324,23 @@ class OpenFileDialogFragment : BaseBottomSheetFragment() {
 
         dialog.setOnShowListener {
             val bottomSheet = (it as BottomSheetDialog)
-                .findViewById<View>(com.google.android.material.R.id.design_bottom_sheet)
+                .findViewById(com.google.android.material.R.id.design_bottom_sheet)
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFileDialogFragment.kt:327]: This should be directly parametrized as FrameLayout (findViewById<FrameLayout>)

**Comment by EmmanuelMess** [comment]
Pressing on the top of the bottom sheet and swiping down doesn't do anything.

**Comment by VishalNehra** [comment]
@EmmanuelMess Yea, basically all gestures now lead to expanded state. Couldn't find any other way to fix this. 
I feel this doesn't effect the UX much as pressing outside the sheet or device back button closes the sheet still.

**Review by EmmanuelMess — APPROVED** [review]

**Comment by EmmanuelMess** [comment]
@VishalNehra codacy failed.

**Comment by EmmanuelMess** [comment]
Lets test if I can trigger a codacy check with an empty commit.

**Pull request merged**